### PR TITLE
[NFC] Refactor getNativeSILFunctionType to use a lambda.

### PR DESCRIPTION
**Pull request by varungandhi-apple**
Minor cleanup.

### CODE REVIEW

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

File: lib/SIL/IR/SILFunctionType.cpp [hunk 1]
```diff
@@ -2377,42 +2383,29 @@ static CanSILFunctionType getNativeSILFunctionType(
     switch (constant ? constant->kind : SILDeclRef::Kind::Func) {
     case SILDeclRef::Kind::Initializer:
     case SILDeclRef::Kind::EnumElement:
-      return getSILFunctionType(TC, context, origType, substInterfaceType,
-                                extInfo, DefaultInitializerConventions(),
-                                ForeignInfo(), origConstant, constant, reqtSubs,
-                                witnessMethodConformance);
+      return getSILFunctionTypeForConventions(DefaultInitializerConventions());
```
Comment (slavapestov) [inline lib/SIL/IR/SILFunctionType.cpp:2386]: Another approach I use in similar situations is to factor out the switch into a new function that in this case would return the conventions, and then call getSILFunctionType() with the result of this new function.

Comment (varungandhi-apple) [reply lib/SIL/IR/SILFunctionType.cpp:2386]: I tried doing that here, but `Conventions` is an abstract class, so I cannot easily return it, I'd have to do some manual boxing+unboxing.

Comment (slavapestov) [reply lib/SIL/IR/SILFunctionType.cpp:2386]: You could return a unique_ptr<Conventions>, but yeah.

**Comment by varungandhi-apple** [comment]
@swift-ci please test Windows

**Pull request merged**